pkg/tiller/release_server.go

```diff
@@ -448,7 +448,7 @@ func (s *ReleaseServer) deleteHookByPolicy(h *release.Hook, policy string, name,
 	b := bytes.NewBufferString(h.Manifest)
 	if hookHasDeletePolicy(h, policy) {
 		s.Log("deleting %s hook %s for release %s due to %q policy", hook, h.Name, name, policy)
-		if errHookDelete := kubeCli.Delete(namespace, b); errHookDelete != nil {
+		if errHookDelete := kubeCli.DeleteWithTimeout(namespace, b, 60, true); errHookDelete != nil {
```

Comment:
Should this timeout value be configurable by annotation on the hook? If we don't want that, then this definitely should be a constant so we can give it a meaningful name

Reply:
I've moved the timeout value into a constant. Making it configurable makes sense, although I think the default should be a timeout larger than 0. Waiting for the resource to be fully deleted is a reasonable default behavior.

So would it make sense to create an additional annotation, something like `helm.sh/hook-delete-timeout` that would override the default value? Or allow appending the timeout value at the end of the current hooks, something like `before-hook-creation:60`? I think I prefer the first alternative.

Reply:
I think I agree with you with what the name should be. Let me check with some of the other maintainers as well.

Reply:
Yeah, let's go with `helm.sh/hook-delete-timeout` and add that as an override for the default. Thanks for the work on this @mortent!